### ISSUE TITLE
 Using relative config files in dicts and lists raises `Unexpected import path format`

### ISSUE THREAD

**Issue by AlejandroBaron** [omni-us/jsonargparse#557]
Using relative config files in dicts doesn't work. It might be that I'm not using the correct syntax tho

### To reproduce
3 files in the same directory

1) `cli.py`
```
from jsonargparse import CLI
from typing import Dict, List

class MyObject:

    def __init__(self, x: int) -> None:
        self.x = x


class MyCLI:

    def __init__(self, d: Dict[str, MyObject]) -> None:
        self.d = d


CLI(MyCLI)
```
(If you use a `List` instead of `Dict` for `d` param, same error happens)

2) `main.yaml`
```
d:
  obj1: ./my_obj.yaml
```
3) `my_obj.yaml`
```
class_path: cli.MyObj
init_args:
  x: 3
```

Command: `python cli.py  --config main.yaml`. Raises
```
error: Parser key "d":
  Unexpected import path format: ./my_obj.yaml
```

As I said it might not be a bug. I'm using python 3.8, but in the project that triggered this error I'm using python 3.9.

**Comment by mauvilsa** [comment]
This is not a bug. Not everything can be parsed as a subconfig. A relative path to a subconfig inside another config only works if the parser has an action that would load it. Subconfigs are only available for top-level classes in a typehint, not nested like in a dict. Generally you can see which subconfigs can be loaded by printing the `--help`.

If not a bug, could this be added as a new feature? Might be, though most likely it would be technically quite complex and be low priority.

**Comment by AlejandroBaron** [comment]
> This is not a bug. Not everything can be parsed as a subconfig. A relative path to a subconfig inside another config only works if the parser has an action that would load it. Subconfigs are only available for top-level classes in a typehint, not nested like in a dict. Generally you can see which subconfigs can be loaded by printing the `--help`.
> 
> If not a bug, could this be added as a new feature? Might be, though most likely it would be technically quite complex and be low priority.

Yeah I assumed that it was not a bug but just not supported right now. Is there a way/plan to use resolvers like omegaconf does for this kind of situation?

**Comment by mauvilsa** [comment]
omegaconf resolvers are able to load subconfigs? I don't see that [here](https://omegaconf.readthedocs.io/en/latest/custom_resolvers.html), but I am not an expert on that. If it is supported, then you can make jsonargparse to use omegaconf as the yaml loader.

**Comment by mauvilsa** [comment]
> Is there a way/plan to use resolvers like omegaconf does for this kind of situation?

Now (v4.41) it is possible to register and use any omegaconf resolver, see in the [docs](https://jsonargparse.readthedocs.io/en/v4.41.0/#experimental-omegaconf-mode). For now it is experimental.

**Comment by mauvilsa** [comment]
Pull request #940 added support for this. Though, note that the reproduction above as is has mistakes and would fail. For it to work, it would need to be:

```python
# cli.py

from jsonargparse import auto_cli
from typing import Dict

class MyObject:
    def __init__(self, x: int) -> None:
        self.x = x

class MyCLI:
    def __init__(self, d: Dict[str, MyObject]) -> None:
        self.d = d
        print({k: v.x for k, v in d.items()})

if __name__ == "__main__":
    auto_cli(MyCLI)
```

```yaml
# my_obj.yaml

class_path: __main__.MyObject
init_args:
  x: 3
```
